flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/FsStateChangelogWriter.java

```diff
@@ -369,7 +390,6 @@ public void confirm(SequenceNumber from, SequenceNumber to, long checkpointId) {
                             changelogRegistry.stopTracking(localHandle);
                             localChangelogRegistry.register(localHandle, checkpointId);
                         });
-        localChangelogRegistry.discardUpToCheckpoint(checkpointId);
```

Comment:
In order to delete useless local files as soon as possible, discardUpToCheckpoint should not be removed from here.

Reply:
Fixed, retain the call of `discardUpToCheckpoint` here and delete the call in `subsume()`.